### DISCUSSION TITLE
"Public key not found" when installing Eclipse using Oomph installer in WSL2

### DISCUSSION THREAD

**Discussion by fedejeanne** [eclipse-platform/eclipse.platform#1363] [Q&A] [answered]
_(Follow up of https://github.com/eclipse-platform/eclipse.platform/discussions/1360#discussioncomment-9417548)_

Hello again,
I managed to start the installation of the [preconfigured Eclipse Development Environment](https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#creating-an-eclipse-development-environment) in WSL2 using the Oomph installer (for more details, see the [previous discussion](https://github.com/eclipse-platform/eclipse.platform/discussions/1360)) but now I'm running into the following errors when hitting the "Finish" button in Oomph:

```
An error was detected while performing the engine operation and the changes are being rolled back. See the log for details.
ERROR: org.eclipse.equinox.p2.engine code=4 An error occurred while collecting items to be installed
  at org.eclipse.oomph.util.OomphPlugin.coreException(OomphPlugin.java:291)
  at org.eclipse.oomph.p2.internal.core.ProfileTransactionImpl$3.commit(ProfileTransactionImpl.java:553)
  at org.eclipse.oomph.p2.internal.core.ProfileTransactionImpl.commit(ProfileTransactionImpl.java:345)
  at org.eclipse.oomph.setup.p2.impl.P2TaskImpl.perform(P2TaskImpl.java:904)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.doPerformNeededSetupTasks(SetupTaskPerformer.java:3851)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performNeededSetupTasks(SetupTaskPerformer.java:3779)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performTriggeredSetupTasks(SetupTaskPerformer.java:3760)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.perform(SetupTaskPerformer.java:3638)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$9.run(ProgressPage.java:595)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$11$1.run(ProgressPage.java:722)
  at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
  ERROR: org.eclipse.equinox.p2.engine code=0 session context was:(profile=_home_visjee_platform-sdk2_eclipse, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
  ERROR: org.eclipse.equinox.p2.artifact.repository code=4 Result of processing steps.
    OK: unknown code=0 OK
    OK: unknown code=0 OK
    ERROR: org.eclipse.equinox.p2.artifact.repository code=0 Public key not found for 6640597757109451687.
    OK: unknown code=0 OK
  ERROR: org.eclipse.equinox.p2.artifact.repository code=4 Result of processing steps.
    OK: unknown code=0 OK
    OK: unknown code=0 OK
    ERROR: org.eclipse.equinox.p2.artifact.repository code=0 Public key not found for -5034155639734203956.
    OK: unknown code=0 OK
  ERROR: org.eclipse.equinox.p2.artifact.repository code=4 Result of processing steps.
...
```

(see the [full_log.txt](https://github.com/eclipse-platform/eclipse.platform/files/15294730/full_log.txt))

I assume this has something to do with missing certificates? If that's the case, were could I find the necessary certificates?

Once I have them, I would try to import them to my Ubuntu (in WSL2) and try again.

Thank you in advance!

**Comment by merks** [comment, marked as answer]
This sounds like an ancient type of problem from before the days that PGP support was functioning correctly.  Furthermore, the stack trace information does not align with the current code base of Oomph suggesting also that you are not running the latest code and hence are likely running with some ancient version of p2.  All this is very odd because the installer generally installs a nightly build or a milestone build. 

Can you share the versions of the Oomph features you have installed and the about dialog of the installation?

And did you really use an up-to-date installer?

**Comment by Bananeweizen** [comment]
both java and oomph are years old, according to the log.

**Reply by merks** [reply]
I didn't notice the log link.  Indeed that answers that question.  🥇

**Reply by fedejeanne** [reply]
You're right, I was using an old version (I found this in the log, thank you @Bananeweizen for the hint!)

```
Bundle org.eclipse.oomph.setup.installer 1.22.0.v20210817-0859, build=5169, branch=4f2d83d895798b31dc987dc9c07da7c94b536f1d
```

Thank you both for your help!
